sound/soc/sof/core.c

```diff
@@ -348,6 +348,9 @@ static int sof_probe(struct platform_device *pdev)
 		goto comp_err;
 	}
 
+	/* save acpi machine descriptor for machine driver */
+	dev_set_drvdata(&plat_data->pdev_mach->dev, plat_data->machine);
+
```

Comment:
better to handle SPI case also, e.g. if (snd_sof_pdata.type == SOF_DEVICE_SPI) then dev_set_drvdata(&plat_data->pdev_mach->dev, plat_data->sof_machine);

Reply:
@RanderWang I think you've found a valid issue but I don't agree with this patch.

There is a difference in the way the machine drivers expect the information to be passed
````text
bxt_pcm512x.c:265:	mach = (&pdev->dev)->platform_data;
bxt_wm8804.c:222:	mach = (&pdev->dev)->platform_data;
bytcht_da7213.c:235:	mach = (&pdev->dev)->platform_data;
bytcht_es8316.c:250:	mach = (&pdev->dev)->platform_data;
bytcr_rt5640.c:1131:	mach = byt_rt5640_card.dev->platform_data;
bytcr_rt5651.c:895:	mach = byt_rt5651_card.dev->platform_data;
cht_bsw_rt5645.c:553:	mach = (&pdev->dev)->platform_data;
cht_bsw_rt5672.c:404:	struct snd_soc_acpi_mach *mach = pdev->dev.platform_data;
skl_hda_dsp_generic.c:150:	mach = dev_get_drvdata(&pdev->dev);
````
So the first order of business is to realign, not to add a new method that only works for skl_hda_dsp_generic.
I overlooked this point when I tested the SST case.

Reply:
yes, I agree with you. First I also found this dev_get_drvdata is used only for skl_hda_dsp but it was upstreamed. If you would refine it, I would wait for your solution.